### PR TITLE
Reduced API calls on System Parameters page

### PR DESCRIPTION
**Pull request by vedangimittal**
- Reduced API calls on System Parameters page for performance improvement
- Jira Story- https://jsw.ibm.com/browse/PFEBMC-2945

### CODE REVIEW

**Comment by vedangimittal** [comment]
Will remove the comments before getting it merged.

File: src/store/modules/ResourceManagement/SystemParametersStore.js [hunk 1]
```diff
@@ -69,138 +74,67 @@ const systemParametersStore = {
       (state.rpdScheduledRunDuration = rpdScheduledRunDuration),
   },
   actions: {
-    async getAggressivePrefetch({ commit }) {
+    async getApi({ commit }) {
```
Comment (Nikhil-Ashoka) [inline src/store/modules/ResourceManagement/SystemParametersStore.js]: Rename: "getApi" to "getBiosAttributesRegistry"

File: src/views/ResourceManagement/SystemParameters/SystemParameters.vue [hunk 2]
```diff
@@ -42,17 +42,18 @@ export default {
   created() {
     this.startLoader();
     Promise.all([
```
Comment (Nikhil-Ashoka) [inline src/views/ResourceManagement/SystemParameters/SystemParameters.vue]: No need of Promise.all as we are making only one API call

File: src/store/modules/ResourceManagement/SystemParametersStore.js [hunk 3]
```diff
@@ -209,24 +143,105 @@ const systemParametersStore = {
...
+            (Attribute) => Attribute.AttributeName == 'hb_lateral_cast_out_mode'
+          );
+          let lateralCastOutModeValue = lateralCastOutMode[0].CurrentValue;
+          let lateralCastModeValue =
+            lateralCastOutModeValue == 'Enabled' ? true : false;
+          commit('setLateralCastOutMode', lateralCastModeValue);
         })
         .catch((error) => console.log(error));
     },
+    // async getAggressivePrefetch({ commit }) {
```
Comment (Nikhil-Ashoka) [inline src/store/modules/ResourceManagement/SystemParametersStore.js]: Remove unused code

File: src/views/ResourceManagement/SystemParameters/SystemParameters.vue [hunk 4]
```diff
@@ -42,17 +42,18 @@ export default {
...
-      this.$store.dispatch('systemParameters/getGuardOnError'),
-      this.$store.dispatch('systemParameters/getRpdPolicyOptions'),
-      this.$store.dispatch('systemParameters/getRpdFeatureOptions'),
-      this.$store.dispatch('systemParameters/getRpdPolicy'),
-      this.$store.dispatch('systemParameters/getRpdPolicyCurrent'),
-      this.$store.dispatch('systemParameters/getRpdFeature'),
-      this.$store.dispatch('systemParameters/getRpdScheduledRun'),
-      this.$store.dispatch('systemParameters/getRpdScheduledRunDuration'),
+      this.$store.dispatch('systemParameters/getApi'),
+      // this.$store.dispatch('systemParameters/getLateralCastOutMode'),
```
Comment (Nikhil-Ashoka) [inline src/views/ResourceManagement/SystemParameters/SystemParameters.vue]: Remove unused code

**Review by Nikhil-Ashoka — CHANGES_REQUESTED** [review]
Please address the comments

File: src/store/modules/ResourceManagement/SystemParametersStore.js [hunk 5]
```diff
@@ -69,138 +74,67 @@ const systemParametersStore = {
...
           );
-          let rpdFeatureOptions = rpdPolicy[0].Value.map(
+          let rpdFeatureOptions = rpdFeatureOps[0].Value.map(
             ({ ValueName }) => ValueName
           );
           commit('setRpdFeatureOptions', rpdFeatureOptions);
-        })
-        .catch((error) => console.log(error));
-    },
-    async getRpdScheduledRun({ commit }) {
```
Comment (Nikhil-Ashoka) [inline src/store/modules/ResourceManagement/SystemParametersStore.js:197]: There is an error. The methods that you have removed, But being called somewhere in the code.

Comment (Nikhil-Ashoka) [reply src/store/modules/ResourceManagement/SystemParametersStore.js:197]: Make sure that the method name that is removed is not being used anywhere in the code. Eg: Click on "Update RPD policy" and there is vuex error in console.
image

File: src/store/modules/ResourceManagement/SystemParametersStore.js [hunk 6]
```diff
@@ -387,21 +319,14 @@ const systemParametersStore = {
...
-        .catch((error) => console.log(error));
-    },
+    // async getLateralCastOutMode({ commit }) {
+    //   const lateralCastOutMode = this.registryEntries.Attributes.filter(
+    //     (Attribute) => Attribute.AttributeName == 'hb_lateral_cast_out_mode'
+    //   );
+    //   let lateralCastOutModeValue = lateralCastOutMode[0].CurrentValue;
+    //   let modeValue = lateralCastOutModeValue == 'Enabled' ? true : false;
+    //   commit('setLateralCastOutMode', modeValue);
+    // },
```
Comment (nabilpandiyalayil) [inline src/store/modules/ResourceManagement/SystemParametersStore.js]: Please remove these lines.

**Review by Nikhil-Ashoka — APPROVED** [review]

**Pull request merged**